packages/fresh_dio/lib/src/fresh.dart

```diff
@@ -74,76 +74,97 @@ class Fresh<T> extends Interceptor with FreshMixin<T> {
   final RefreshToken<T> _refreshToken;
 
   @override
-  Future<dynamic> onRequest(RequestOptions options) async {
+  Future<dynamic> onRequest(
+    RequestOptions options,
+    RequestInterceptorHandler handler,
+  ) async {
     final currentToken = await token;
     final headers = currentToken != null
         ? _tokenHeader(currentToken)
         : const <String, String>{};
     options.headers.addAll(headers);
-    return options;
+    handler.next(options);
   }
 
   @override
-  Future<dynamic> onResponse(Response response) async {
+  Future<dynamic> onResponse(
+    Response response,
+    ResponseInterceptorHandler handler,
+  ) async {
     if (await token == null || !_shouldRefresh(response)) {
-      return response;
+      return handler.next(response);
+    }
+    try {
+      final refreshResponse = await _tryRefresh(response);
+      handler.resolve(refreshResponse);
+    } on DioError catch (error) {
+      handler.reject(error);
     }
-    return _tryRefresh(response);
   }
 
   @override
-  Future<dynamic> onError(DioError err) async {
+  Future<dynamic> onError(
+    DioError err,
+    ErrorInterceptorHandler handler,
+  ) async {
     final response = err.response;
-    if (await token == null ||
+    if (response == null ||
+        await token == null ||
         err.error is RevokeTokenException ||
         !_shouldRefresh(response)) {
-      return err;
+      return handler.next(err);
+    }
+    try {
+      final refreshResponse = await _tryRefresh(response);
+      handler.resolve(refreshResponse);
+    } on DioError catch (error) {
+      handler.next(error);
     }
-    return _tryRefresh(response);
   }
 
-  Future<dynamic> _tryRefresh(Response? response) async {
-    T refreshedToken;
+  Future<Response<dynamic>> _tryRefresh(Response response) async {
+    late final T refreshedToken;
     try {
+      _httpClient.lock();
       refreshedToken = await _refreshToken(await token, _httpClient);
```

Comment:
It seem that my refresh requests are not sent anymore after updating. I assume this is caused by the lock. The `_httpClient` instance is locked so I have to use another instance to make the refresh request?